### ISSUE TITLE
PENDING: Operation enqueued from HTTP endpoint

### ISSUE THREAD

**Issue by d-shi** [linkedin/cruise-control#168]
I'm trying to get started with CC. I added `metric.reporters=com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter` along with the corresponding jar to our broker. I can see that it has created the topics `__CruiseControlMetrics-0`, `__KafkaCruiseControlModelTrainingSamples-{0..31}`, and `__KafkaCruiseControlPartitionMetricSamples-{0..31}`.

I don't see any errors in the broker logs or the CC logs. When trying to access the `/kafkacruisecontrol/state` HTTP endpoint in the browser, I just get:

```
( 10001 ms) - (100.0%) PENDING: Operation enqueued, waiting to be executed.
```

Any ideas what I'm doing wrong?

**Comment by becketqin** [comment]
We have enabled session support in CC. If an operation needs to take a while to finish, CC will return a progress after some time (by default 10 seconds). This is essentially what you saw. If you query again, you will see the progress of you previous request associated with your session. But if you wait too long before issuing the next query, your session may be timed out and a new request will be created.

**Comment by d-shi** [comment]
If I keep refreshing the page the number of ms on the left goes up, but I still get the same message. Does 100% mean it should be finished though? Right now there's basically 0 data to load since this is a brand new test cluster I'm experimenting with which has only 1 broker. I don't think it should be taking this long to process the state endpoint...

**Comment by becketqin** [comment]
@d-shi Yeah, that does sound a bug. Thanks for reporting this. Let me see if I can reproduce.

**Comment by d-shi** [comment]
@becketqin I managed to get it working. I suspect my issue had to do with some networking setup, as I was running the kafka broker and zookeeper inside kubernetes pods, but CC on my dev machine and connecting to the broker/zk through `kubectl port-forward`. After building my own docker image for CC and deploying it inside kubernetes it seems to work now.
